Block settings dropdown: use block display title in remove label

ramonjd opened the pull request:
## Description
This PR employs a block display title in the block settings dropdown rather than the default core block name in the drop down label "Remove `x` block".

A "block display title" might be a specific block variation title, or a reusable block group name. 

This makes the label consistent with the block list view.

## Changes
- Gutting `<BlockTitle />` and turning it into a hook that we use independently of JSX
- Truncating long block display names to 25 chars (negotiable) ONLY for `BlockSettingsDropdown`, which appears in the [block settings toolbar](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-toolbar/index.js#L145) and the [block list view](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/list-view/block.js#L248). This is to maintain (roughly) the dropdown container's current width proportions.
- Making truncation of the block title optional. I've passed the current default value of `35` for all existing usages to maintain the status quo for now.

## Room for improvement

We might like to refactor `<Blocktitle />` to use the `<Text />` component, which has truncation built in, e.g., 

```jsx
<Text
    as="span"
    limit={ 25 }
    ellipsizeMode="tail"
    truncate
    >
   { someLabel }
</Text>
```

Also, maybe an aria-label with the full value might be appropriate where there is a truncation?

## Testing Instructions

1. Add a Group Block to a post.
2. Add a Row Block to a post.
3. Add another random block to a post, and turn it into a reusable block.
4. When removing the block via the block settings drop down, either via the Block List Tree ellipsis, or via the block's own toolbar,  you should see the name of either the block, or its variation or the reusable group.
5. Check that other instances of `<BlockTitle />` appear as they do on trunk (max 35 chars), e.g., bread crumb, block switcher...


Run `npm run test-unit packages/block-editor/src/components/block-title/test/index.js`

<details><summary>Some test code</summary>

```html
<!-- wp:group {"layout":{"type":"flex","allowOrientation":false}} -->
<div class="wp-block-group"></div>
<!-- /wp:group -->

<!-- wp:group -->
<div class="wp-block-group"></div>
<!-- /wp:group -->
```

</details>

## Screenshots <!-- if applicable -->
### Before

Remove a "Row" Block (a Group variation)

<img width="555" alt="Screen Shot 2022-02-28 at 1 52 29 pm" src="https://user-images.githubusercontent.com/6458278/155916226-ee65b20b-c5b3-4dbc-8603-561d178f1c5e.png">

Remove a reusable block 
<img width="537" alt="Screen Shot 2022-02-28 at 1 52 25 pm" src="https://user-images.githubusercontent.com/6458278/155916231-6d5a55ad-1563-45b1-9a54-84545720d378.png">


### After
<img width="578" alt="Screen Shot 2022-02-28 at 1 13 04 pm" src="https://user-images.githubusercontent.com/6458278/155916137-5e1c73ce-4f93-49b2-b52a-aa1cd23e149d.png">

<img width="675" alt="Screen Shot 2022-02-28 at 1 47 43 pm" src="https://user-images.githubusercontent.com/6458278/155916116-5c3ffc17-06e8-4131-ada7-015a3a7458f0.png">




## Types of changes
Enhancement to the block settings dropdown.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->
- [ ] I've updated related schemas if appropriate. <!-- Reference: https://github.com/WordPress/gutenberg/tree/trunk/schemas -->

ramonjd committed:
Use a block display title rather than the default core block name in the drop down label, similar to the way the block list view does it.
Add a maximum truncate value
Updated tests

ramonjd committed:
Updating docs

andrewserong commented on packages/block-editor/README.md:
```diff
@@ -228,10 +228,11 @@ _Parameters_
 _Returns_
 
--   `?string`: Block title.
+-   `JSX.Element`: Block title.
```
Ah, good catch. I did a quick double-take on that one, because the _component_ is returning a `string`, but of course React renders it as a `JSX.Element`, so that's a more accurate description of what it's returning. Is that right?

ramonjd replied:
Yes, nice spotting. Actually this was an automated update. 

I left it there since the component is currently used as a component, so `<BlockTitle />`. It's used by the Block List Tree in this way for example.

But of course `BlockTitle( { clientId } )` will return a plain string.

I did hesitate before splitting the component into component + hook and just calling `BlockTitle( { clientId } )`, but perceived convention prevented me.

What do you reckon?

ramonjd replied:
Another option that occurred to me would be to refactor `<BlockTitle />` to be a regular function.

It _ostensibly_ appears to be a component, but only due to the way it's being used.

Given that it doesn't return any markup maybe it's a possible follow up. 🤷

andrewserong replied:
> Actually this was an automated update.

Ah, nice!

> What do you reckon?

I think I'd lean toward the approach you've gone with in this PR — abstract away the internals of that component so that you've got reusability for the case of the block settings dropdown, then we could look at refactoring `BlockTitle` separately if need be. I suppose there's nice ergonomics of adding it to a component via `<BlockTitle clientId={ clientId } />` rather than having to call a function within JSX like `{ getBlockTitle( clientId ) }`?. Also, it looks like the `BlockTitle` component is used in half a dozen places, so it might be better to leave it as a component (at least for the time being) 🙂

ramonjd replied:
Great, thanks for your thoughts. Sounds very reasonable 👍

andrewserong commented on packages/block-editor/src/components/block-title/use-block-display-title.js:
````diff
@@ -0,0 +1,87 @@
+ * @example
+ *
+ * ```jsx
+ * <BlockTitle clientId="afd1cb17-2c08-4e7a-91be-007ba7ddc3a1" />
````
Since this has been moved to a hook, should the example go in `index.js` instead?

ramonjd replied:
Another win for copy 🍝 

Thanks, I'll remove 👍

andrewserong reviewed (approved):
Nice one @ramonjd — good idea abstracting the logic from the `BlockTitle` component and moving it to a hook with custom truncation length 👍.

This is testing nicely for me in the post and site editors, the List View display of the block title still looks good, and now the text for removing a Row block looks correct, along with removing a Reusable block:

| Removing a Row block | Removing a Reusable block |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/155920676-7e0b233b-af1a-48dd-8a51-97ff4c4eaf72.png) | ![image](https://user-images.githubusercontent.com/14988353/155920691-3650b435-f98c-4544-aa3d-1dd15e6a109b.png) |

I just left a tiny comment about where we use an example of the `BlockTitle` component, but this LGTM! Might be worth getting another pair of eyes on it too, just in case 🙂

ramonjd committed:
Updating inline docs

ramonjd committed:
Updating docs

gziolo commented:
> Truncating long block display names to 25 chars (negotiable)

It looks lit was updated to 35 chars. Anyway, I'm slightly concerned about using any default value for truncating the computed block's title. It will impact every usage of the `<BlockTitle />` component, including other places of the editor like the breadcrumb in the footer or the usage in the React Native mobile app. Even if the decision would be to use this default value, it would be great to update the documentation of the Block API and recommend shorter titles for blocks. It looks like this impacts also titles created by users in the UI for the Reusable block or Template Part block.

ramonjd commented:
Thanks for checking it out @gziolo ! 🙇 

> It looks lit was updated to 35 chars. 

I left the default of 35 alone. It was bumped from [15 more than a year ago](https://github.com/WordPress/gutenberg/pull/28691/files#diff-fc3dd7fb0a590fa2e6c702e97b63d14a08338a9031064257b21dd7b730c2acadL74).

The `25` characters introduced in this PR is only for the block setting drop down list to maintain (roughly) the dropdown container's current width proportions. Apologies, I probably didn't make that very clear in the description. I've updated it.

> It looks like this impacts also titles created by users in the UI for the Reusable block or Template Part block.

The `BlockSettingsDropdown` appears in the [block settings toolbar](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-toolbar/index.js#L145) and the [block list view](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/list-view/block.js#L248), so any block titles displayed in that drop down.

The goal was to avoid huge titles in the drop down and therefore very wide dropdown panels.  

| 35  chars | 25 chars |
| ------------- | ------------- |
| <img width="656" alt="Screen Shot 2022-02-28 at 1 22 27 pm" src="https://user-images.githubusercontent.com/6458278/155948253-98d77ca1-d65f-413c-8c54-690b519200fd.png">  | <img width="675" alt="Screen Shot 2022-02-28 at 1 47 43 pm" src="https://user-images.githubusercontent.com/6458278/155948281-57da6d69-314a-4468-ad40-c71c55a4b8a6.png"> |



> Anyway, I'm slightly concerned about using any default value for truncating the computed block's title. It will impact every usage of the <BlockTitle /> component, including other places of the editor like the breadcrumb in the footer or the usage in the React Native mobile app. 

Agree that truncation is probably not appropriate in all cases. For the Block List Tree and, arguably, the drop down menus, it might be desirable to defend against loquacious labels.

I'm happy to create another PR to audit the usage of `BlockTitle` and pass a maximum string value only where it would make sense from a UX/readability perspective. 

It might be worthwhile adding an `aria-label` in the case where it's truncated as well. I wanted to do that in this PR but was trying keep things focussed and review-friendly.

> Even if the decision would be to use this default value, it would be great to update the documentation of the Block API and recommend shorter titles for blocks. 

As part of any follow up I can definitely update the documentation of the Block API to recommended shorter titles in general for blocks. Sounds like a very reasonable thing to communicate to folks.

talldan commented on packages/block-editor/src/components/block-title/use-block-display-title.js:
````diff
@@ -0,0 +1,87 @@
+ * useBlockDisplayTitle( 'afd1cb17-2c08-4e7a-91be-007ba7ddc3a1', 17 );
+ * ```
+ *
+ * @param {string} clientId Client ID of block.
+ * @param {number} maximumLength The maximum length that the block title string may be before truncated.
+ * @return {?string} Block title.
+ */
+export default function useBlockDisplayTitle( clientId, maximumLength ) {
+	maximumLength = Number.isInteger( maximumLength )
+		? maximumLength
+		: MAXIMUM_TITLE_LENGTH;
````
Could `maximumLength` be treated as an optional param with a default value?
```suggestion
 * @param {string} clientId Client ID of block.
 * @param {number|undefined} maximumLength The maximum length that the block title string may be before truncated.
 * @return {?string} Block title.
 */
export default function useBlockDisplayTitle( clientId, maximumLength = MAXIMUM_TITLE_LENGTH ) {
```

talldan replied:
Now that I've said that I've noticed that there doesn't seem to be a way to always get the full block title without any truncation. The only option is to use a really big number for the max length.

Because a new API is being introduced with this hook there's an opportunity to step away from the way `BlockTitle` did things.

The truncation in `BlockTitle` could still be left working as it was for back compat by passing in a max length param to the hook, but maybe when `undefined` is passed to the hook it should return the full block title (and not have an implicit default at all).

gziolo replied:
Right, that was my initial thinking that isn't possible to show the full title with the revised implementation. There is some legacy handling here that truncates long titles for reusable blocks but it wasn't applied to other blocks. So maybe we could keep it as the default, and truncate everything when an explicit value is passed.

ramonjd replied:
Thanks for your help @talldan and @gziolo 

> maybe when undefined is passed to the hook it should return the full block title (and not have an implicit default at all
> So maybe we could keep it as the default, and truncate everything when an explicit value is passed.

 I agree that it makes a lot of sense if truncation is optional. 

I've taken this route in f625f28f8d6b65f6a0bc181874801eb5fc85c35c since it allows for greater flexibility. 👍 

For all existing usages I'm passing `35` (the current default) for now to maintain the status quo and to keep this PR manageable. 

If we want to revisit individual lengths, I'm happy to do it in a follow up PR. How does that sound?

talldan commented on packages/block-editor/src/components/block-title/index.js:
```diff
@@ -26,55 +13,18 @@ import { store as blockEditorStore } from '../../store';
-		return truncate( label, { length: 35 } );
-	}
-	return blockInformation.title;
+export default function BlockTitle( {
+	clientId,
+	maximumLength = MAXIMUM_TITLE_LENGTH,
+} ) {
+	return useBlockDisplayTitle( clientId, maximumLength );
```
I think there's no need to specify a default here as `useBlockDisplayTitle` seems to already fall back to the default maximum length implicitly when `undefined` is passed:
```suggestion
export default function BlockTitle( {
	clientId,
	maximumLength,
} ) {
	return useBlockDisplayTitle( clientId, maximumLength );
```

That should hopefully mean it will be possible to remove the import of `MAXIMUM_TITLE_LENGTH`.

ramonjd committed:
Remove truncation by default from <BlockTitle />
Pass `35` as maximumLength to maintain backwards compatibility for current usages
Updated tests

ramonjd committed:
I didn't mean to check that in :)

ramonjd commented:
Updated in f625f28f8d6b65f6a0bc181874801eb5fc85c35c to remove truncation by default. 

I've preserved the current, default truncation value of `35` for current usages of `<BlockTitle />`.  For example, in the bread crumb

<img width="400" alt="Screen Shot 2022-03-04 at 10 02 24 pm" src="https://user-images.githubusercontent.com/6458278/156754478-615775e3-2b6f-4db4-9300-a02cf8f5e6a6.png">

talldan commented on packages/block-editor/src/components/block-title/use-block-display-title.js:
```diff
@@ -0,0 +1,83 @@
+	// Label will fallback to the title if no label is defined for the current
+	// label context. If the label is defined we prioritize it over possible
+	// possible block variation title match.
+	if ( label && label !== blockType.title ) {
+		return maximumLength && maximumLength > 0
+			? truncate( label, { length: maximumLength } )
+			: label;
+	}
```
It's minor, but it might be worth mentioning in the docs for the `maximumLength` param that truncation only applies to the `label` (and not the `title`).

I'm not sure the best way to explain it, maybe mentioning that truncation only applies when the block displays something other than its title (e.g. a variation name or a custom label).

ramonjd replied:
Good point. Definitely worth communicating that. I'll throw up a small PR.

ramonjd replied:
https://github.com/WordPress/gutenberg/pull/39328

gziolo replied:
What if the block title has 100 characters? 😄 

Maybe we should also truncate it for consistency when someone explicitly passes this parameter.

ramonjd replied:
> What if the block title has 100 characters? 😄

💥 

Haha. Yeah, the expectation seems to have been that block title strings should be protected from truncation or were never going to be long enough to be truncated.

> Maybe we should also truncate it for consistency when someone explicitly passes this parameter.

A cursory grep through the `block.json` files (there might be a few e2e fixtures in there) shows that there's not much over 28 chars in length, so we could probably get away with it.

<details><summary>Block title string lengths</summary>

```bash
Read More: 9
Site Logo: 9
Post Author: 11
Video: 5
Post Comments Link: 18
Comment Author Avatar: 21
Custom Link: 11
Comments Pagination: 19
Tag Cloud: 9
Calendar: 8
Buttons: 7
Post Date: 9
Comments Query Loop: 19
Embed: 5
Post Excerpt: 12
File: 4
Paragraph: 9
Quote: 5
Group: 5
Site Tagline: 12
Shortcode: 9
Comment Edit Link: 17
Separator: 9
Gallery: 7
Heading: 7
Comment Template: 16
Comment Content: 15
Term Description: 16
RSS: 3
Post Author Name: 16
Next Page: 9
Page Numbers: 12
Code: 4
Navigation: 10
Verse: 5
Post Comments Count: 19
Pullquote: 9
Post Author Biography: 21
Query Title: 11
Custom HTML: 11
Column: 6
Template Part: 13
Archives: 8
Social Icons: 12
Unsupported: 11
Spacer: 6
Social Icon: 11
Post Navigation Link: 20
Image: 5
More: 4
Table of Contents: 17
Search: 6
Page Break: 10
Text Columns (deprecated): 25
Comment Date: 12
Audio: 5
Post Comments Form: 18
Next Page: 9
Button: 6
Login/out: 9
Table: 5
Post Template: 13
List: 4
Preformatted: 12
Pagination: 10
Classic: 7
Post Terms: 10
Previous Page: 13
Navigation Area: 15
Home Link: 9
Post Featured Image: 19
Page List: 9
Previous Page: 13
Comment Author Name: 19
Columns: 7
Latest Comments: 15
Latest Posts: 12
Pattern: 7
Media & Text: 12
Post Content: 12
Post Title: 10
Query Loop: 10
Site Title: 10
Comment Reply Link: 18
Submenu: 7
Page Numbers: 12
Categories: 10
Post Comments: 13
Reusable block: 14
Post Comment (deprecated): 25
Cover: 5
Iframed Block: 13
Iframed Masonry Block: 21
Iframed Multiple Stylesheets: 28
Iframed Inline Styles: 21
Legacy Widget: 13
Notice: 6
Read More: 9
Site Logo: 9
Post Author: 11
Video: 5
Post Comments Link: 18
Comment Author Avatar: 21
Custom Link: 11
Comments Pagination: 19
Tag Cloud: 9
Calendar: 8
Buttons: 7
Post Date: 9
Comments Query Loop: 19
Embed: 5
Post Excerpt: 12
File: 4
Paragraph: 9
Quote: 5
Group: 5
Site Tagline: 12
Shortcode: 9
Comment Edit Link: 17
Separator: 9
Gallery: 7
Heading: 7
Comment Template: 16
Comment Content: 15
Term Description: 16
RSS: 3
Post Author Name: 16
Next Page: 9
Page Numbers: 12
Code: 4
Navigation: 10
Verse: 5
Post Comments Count: 19
Pullquote: 9
Post Author Biography: 21
Query Title: 11
Custom HTML: 11
Column: 6
Template Part: 13
Archives: 8
Social Icons: 12
Unsupported: 11
Spacer: 6
Social Icon: 11
Post Navigation Link: 20
Image: 5
More: 4
Table of Contents: 17
Search: 6
Page Break: 10
Text Columns (deprecated): 25
Comment Date: 12
Audio: 5
Post Comments Form: 18
Next Page: 9
Button: 6
Login/out: 9
Table: 5
Post Template: 13
List: 4
Preformatted: 12
Pagination: 10
Classic: 7
Post Terms: 10
Previous Page: 13
Navigation Area: 15
Home Link: 9
Post Featured Image: 19
Page List: 9
Previous Page: 13
Comment Author Name: 19
Columns: 7
Latest Comments: 15
Latest Posts: 12
Pattern: 7
Media & Text: 12
Post Content: 12
Post Title: 10
Query Loop: 10
Site Title: 10
Comment Reply Link: 18
Submenu: 7
Page Numbers: 12
Categories: 10
Post Comments: 13
Reusable block: 14
Post Comment (deprecated): 25
Cover: 5
Iframed Block: 13
Iframed Masonry Block: 21
Iframed Multiple Stylesheets: 28
Iframed Inline Styles: 21
Legacy Widget: 13
Read More: 9
Post Comment Author: 19
Site Logo: 9
Post Author: 11
Post Comment Date: 17
Video: 5
Post Comments Link: 18
Comment Author Avatar: 21
Comment Author Avatar: 21
Custom Link: 11
Comments Pagination: 19
Tag Cloud: 9
Calendar: 8
Buttons: 7
Post Date: 9
Comments Query Loop: 19
Embed: 5
Post Comment Content: 20
Post Excerpt: 12
File: 4
Paragraph: 9
Quote: 5
Group: 5
Site Tagline: 12
Shortcode: 9
Post Comment Reply Link: 23
Comment Edit Link: 17
Separator: 9
Gallery: 7
Heading: 7
Comment Template: 16
Comment Content: 15
Term Description: 16
RSS: 3
Post Author Name: 16
Next Page: 9
Page Numbers: 12
Code: 4
Navigation: 10
Verse: 5
Post Comments Count: 19
Pullquote: 9
Post Author Biography: 21
Query Title: 11
Custom HTML: 11
Column: 6
Template Part: 13
Archives: 8
Social Icons: 12
Unsupported: 11
Spacer: 6
Social Icon: 11
Post Navigation Link: 20
Image: 5
More: 4
Table of Contents: 17
Search: 6
Page Break: 10
Text Columns (deprecated): 25
Comment Date: 12
Audio: 5
Post Comments Form: 18
Next Page: 9
Button: 6
Login/out: 9
Table: 5
Post Template: 13
List: 4
Preformatted: 12
Pagination: 10
Classic: 7
Post Terms: 10
Previous Page: 13
Navigation Area: 15
Home Link: 9
Post Featured Image: 19
Page List: 9
Previous Page: 13
Comment Author Name: 19
Post Empty: 10
Post Comment Edit Link: 22
Columns: 7
Latest Comments: 15
Latest Posts: 12
Pattern: 7
Media & Text: 12
Post Content: 12
Post Title: 10
Query Loop: 10
Site Title: 10
Comment Reply Link: 18
Submenu: 7
Page Numbers: 12
Categories: 10
Post Comments: 13
Reusable block: 14
Post Comment (deprecated): 25
Cover: 5
Legacy Widget: 13
Notice: 6
Read More: 9
Site Logo: 9
Post Author: 11
Video: 5
Post Comments Link: 18
Comment Author Avatar: 21
Custom Link: 11
Comments Pagination: 19
Tag Cloud: 9
Calendar: 8
Buttons: 7
Post Date: 9
Comments Query Loop: 19
Embed: 5
Post Excerpt: 12
File: 4
Paragraph: 9
Quote: 5
Group: 5
Site Tagline: 12
Shortcode: 9
Comment Edit Link: 17
Separator: 9
Gallery: 7
Heading: 7
Comment Template: 16
Comment Content: 15
Term Description: 16
RSS: 3
Post Author Name: 16
Next Page: 9
Page Numbers: 12
Code: 4
Navigation: 10
Verse: 5
Post Comments Count: 19
Pullquote: 9
Post Author Biography: 21
Query Title: 11
Custom HTML: 11
Column: 6
Template Part: 13
Archives: 8
Social Icons: 12
Unsupported: 11
Spacer: 6
Social Icon: 11
Post Navigation Link: 20
Image: 5
More: 4
Table of Contents: 17
Search: 6
Page Break: 10
Text Columns (deprecated): 25
Comment Date: 12
Audio: 5
Post Comments Form: 18
Next Page: 9
Button: 6
Login/out: 9
Table: 5
Post Template: 13
List: 4
Preformatted: 12
Pagination: 10
Classic: 7
Post Terms: 10
Previous Page: 13
Navigation Area: 15
Home Link: 9
Post Featured Image: 19
Page List: 9
Previous Page: 13
Comment Author Name: 19
Columns: 7
Latest Comments: 15
Latest Posts: 12
Pattern: 7
Media & Text: 12
Post Content: 12
Post Title: 10
Query Loop: 10
Site Title: 10
Comment Reply Link: 18
Submenu: 7
Page Numbers: 12
Categories: 10
Post Comments: 13
Reusable block: 14
Post Comment (deprecated): 25
Cover: 5
```
</details>

talldan replied:
If the truncation is applied to both the block title and the other labels, that's probably the point at which it doesn't need to be part of the hook.

The same code would just be:
```js
const title = useBlockDisplayTitle( clientId );
const truncatedTitle = truncate( title, maxLength );
```

gziolo replied:
> A cursory grep through the block.json files (there might be a few e2e fixtures in there) shows that there's not much over 28 chars in length, so we could probably get away with it.

Thank you for checking that in WordPress core. It looks like we can postpone the decision 👍🏻 

It's not that I want to push for truncating block titles, but I had this thought that I wanted to share here. When you translate titles they might be longer. As far as I remember the German language has this tendency to be much longer than what you see in English.

ramonjd replied:
> As far as I remember the German language has this tendency to be much longer than what you see in English.

That is an excellent point. 

We might one day have a _Rindfleischetikettierungsüberwachungsaufgabenübertragungsgesetz_ Block and be in real trouble. 😄 

> If the truncation is applied to both the block title and the other labels, that's probably the point at which it doesn't need to be part of the hook.

I think it's probably worth doing to make things work more logically for our future selves, thanks for the example too. I'll make a note and throw up a PR next week.

gziolo replied:
> Rindfleischetikettierungsüberwachungsaufgabenübertragungsgesetz

I see it's a real thing 🙃

ramonjd replied:
> I'll make a note and throw up a PR next week

https://github.com/WordPress/gutenberg/pull/39416

> that's probably the point at which it doesn't need to be part of the hook.

I kept it in the hook for now to keep changes small as `<BlockTitle />` (and therefore the legacy `35` char limit) is used very liberally.

talldan reviewed (approved):
Thanks for making those late changes.

This works really well, and I think it's also a nice little accessibility improvement that the label for the menu item now matches how the item in List View is announced.

andrewserong commented:
In @ramonjd's absence, I'll merge this one in now since it looks like the main feedback has been addressed. We can always tweak further in follow-ups if need be, so let us know here if there was anything else folks wanted to see changed!
